### DISCUSSION TITLE
KIC and Kong Gateway deployment using kgo with postgres

### DISCUSSION THREAD

**Discussion by satngp** [Kong/kong-operator#455] [Q&A]
We are planning to deploy the Kong gateway operator to our cluster with the latest version kgo (1.3.x). We want to deploy the Kong ingress controller (3.2.x) and Kong gateway (3.7.x) both using kgo with Postgres database. Someone, please guide me on the best approach to this. (I see hybrid mode deployment but am not sure how to deploy the latest version of KIC, where in the CP template spec I do not see any specific KIC version.)

**Comment by pmalek** [comment]
Hi @satngp 👋 

You can find how to set the CP image in here: https://github.com/Kong/gateway-operator/blob/c51b941edf2133e4e6dadac4c9e4a0e2c9994295/config/samples/controlplane.yaml#L17.

As for the postgres usage: we advise against that in Kubernetes environments and to use db-less with [Gateway Discovery](https://docs.konghq.com/kubernetes-ingress-controller/3.2.x/production/deployment-topologies/gateway-discovery/).

If you really want, you can make it work by configuring the required env variables etc but this is not something we endorse.

As stated in https://docs.konghq.com/kubernetes-ingress-controller/latest/production/deployment-topologies/db-backed/:

> Database backed deployments should only be used in a small set of circumstances. We recommend using [Gateway Discovery](https://docs.konghq.com/kubernetes-ingress-controller/3.2.x/production/deployment-topologies/gateway-discovery/) unless you’ve been otherwise advised by a member of the Kong team.

**Reply by satngp** [reply]
Hi @pmalek  - Thanks for the quick reply. 

[gateway-operator/config/samples/controlplane.yaml](https://github.com/Kong/gateway-operator/blob/c51b941edf2133e4e6dadac4c9e4a0e2c9994295/config/samples/controlplane.yaml#L17)

Above one is dbless deployment. This one I am able to achieve.  

Is there any sample for dbbacked deployment (with KIC and postgres )

**Reply by pmalek** [reply]
Since db-backed is not endorsed we do not provide the sample manifests for that topology.

**Reply by satngp** [reply]
@pmalek  - thanks. Is there any template I can refer for dbbacked topology to create KIC control plane and Dataplane.

**Reply by pmalek** [reply]
Not really. What you could try is to install `kong` chart with these values https://github.com/Kong/charts/blob/main/charts/kong/example-values/minimal-k4k8s-with-kong-enterprise.yaml and try to reverse engineer the envs that are used in the deployments.

As I mentioned: this is not an approach that Kong endorses in Kubernetes.
